pkg/typesystem/resolver.go

```diff
@@ -19,73 +19,109 @@ const (
 
 // TypesystemResolverFunc is a function that implementations can implement to provide lookup and
 // resolution of a Typesystem.
-type TypesystemResolverFunc func(ctx context.Context, storeID, modelID string) (*TypeSystem, error)
+type TypesystemResolverFunc func(ctx context.Context, storeID, modelID string, validate bool) (*TypeSystem, error)
 
-// MemoizedTypesystemResolverFunc returns a TypesystemResolverFunc that either fetches the provided authorization
-// model (if provided) or looks up the latest authorization model, and then it constructs a TypeSystem from
-// the resolved model. The type-system resolution is memoized so if another lookup of the same model occurs,
-// then the earlier TypeSystem that was constructed will be used.
-//
-// The memoized resolver function is safe for concurrent use.
-func MemoizedTypesystemResolverFunc(datastore storage.AuthorizationModelReadBackend) TypesystemResolverFunc {
-	lookupGroup := singleflight.Group{}
-
-	cache := ccache.New(ccache.Configure[*TypeSystem]())
+type TypesystemResolver interface {
+	ResolveTypesystem(ctx context.Context, storeID, modelID string, validate bool) (*TypeSystem, error)
+}
```

Comment:
Why an interface if, today, there is only one possible implementation? What other implementations do you foresee?

Reply:
We can drop the interface in favor of the explicit `TypesystemResolverFunc`. They just as easily could be used interchangeably, though the interface allows for more mock generation.